app/js/controllers/main.js

```diff
@@ -104,22 +82,70 @@ export default class MainController extends Controller {
       this.appendChildController.mainController = this;
       this.copyMoveController.mainController = this;
 
-      console.log('[Customizer] Lazy-initialized modules');
+      console.log('[Customizer] Initialized controllers and views');
       resolve();
     });
+  }
+
+  _cleanUp() {
+    window.removeEventListener('visibilitychange', this._visibilitychangeHandler);
+
+    this.editController.mainController = null;
+    this.viewSourceController.mainController = null;
+    this.appendChildController.mainController = null;
+    this.copyMoveController.mainController = null;
+
+    this.editController.view = null;
+    this.viewSourceController.view = null;
+    this.appendChildController.view = null;
+    this.copyMoveController.view = null;
+
+    this.editController = null;
+    this.viewSourceController = null;
+    this.appendChildController = null;
+    this.copyMoveController = null;
```

Comment:
Maybe instead we should fire a custom event on the DOM which signals cleanup and that all controllers and views handle. They would clean themselves up and unregister that listener.

Though, since this is already done, I think we should just keep this as-is.